lib/BloqadeQMC/Project.toml

```diff
@@ -20,6 +20,7 @@ RandomNumbers = "e6cf234a-135c-5ec9-84dd-332b85af5143"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+Jackknife = "133c4774-c510-5f25-b650-129312f89f69"
```

Comment:
we shouldn't relay on an external implementation to test our implementation, otherwise we should just use the external implementation.

Reply:
If there are other implementations of all the functions, like jackknife and bootstrap already implemented why not use them?

Reply:
I think I'd prefer to keep the current implementation since it's not hard to test directly (https://github.com/ararslan/Jackknife.jl/blob/main/test/runtests.jl) so that we can work towards stage 1 faster (improving code coverage), otherwise, we may need to wrap/contribute PRs to finish it.

On the other hand, I think @emerali 's implementation seems a bit more generic, but I'm not sure this is what is actually needed. 

In the long run, tho, contributing to existing packages and use them is definitely preferred.

Reply:
Changed the existing jackknife tests to not rely on the external Jackknife package.

For simplicity, also simply adapted the bootstrap tests to jackknife.

Reply:
@Roger-luo

Reply:
That's fine, I think now you can remove this dependency? rest LGTM